flink-runtime/src/main/java/org/apache/flink/runtime/jobgraph/SavepointRestoreSettings.java

```diff
@@ -163,14 +163,16 @@ public static SavepointRestoreSettings forPath(
     public static void toConfiguration(
             final SavepointRestoreSettings savepointRestoreSettings,
             final Configuration configuration) {
-        configuration.set(
-                SavepointConfigOptions.SAVEPOINT_IGNORE_UNCLAIMED_STATE,
-                savepointRestoreSettings.allowNonRestoredState());
-        configuration.set(
-                SavepointConfigOptions.RESTORE_MODE, savepointRestoreSettings.getRestoreMode());
-        final String savepointPath = savepointRestoreSettings.getRestorePath();
-        if (savepointPath != null) {
-            configuration.setString(SavepointConfigOptions.SAVEPOINT_PATH, savepointPath);
+        if (!savepointRestoreSettings.equals(SavepointRestoreSettings.none())) {
```

Comment:
I think the method contains the logic about setting savepointRestoreSettings to Configuration even for SavepointRestoreSettings.none.
Some callers may just want to use SavepointRestoreSettings.none to override the configuration. e.g. RemoteStreamEnvironment#getEffectiveConfiguration (Of course, it seems not so correct about its logic).
So How about letting the caller decide whether toConfiguration ? WDYT?

Reply:
Yes I also notice this behavior, it seems a little bit weird, since it will only write the config of `SAVEPOINT_IGNORE_UNCLAIMED_STATE` and `RESTORE_MODE` in the method `toConfiguration` if we pass a SavepointRestoreSettings.none to it, the savepoint path will not write to the config.

Reply:
Yeah, I think it's also fair to filter SavepointRestoreSettings.none.
IIUC, RemoteStreamEnvironment may also suffer from this before this pr?
So let's also test and remove the strange logic if right.

Reply:
@masteryhx
I agree with your suggestion, and let the caller decide to set the default savepoint restore settings to configuration. 

`RemoteStreamEnvironment` also has this problem, I have changed the related code too, and add unit testes to verify it.